src/Lecture4.hs

```diff
@@ -154,7 +173,10 @@ string.
 If both strings have the same length, return the first one.
 -}
 instance Semigroup MaxLen where
-
+  a <> b = case compare (length (unMaxLen a)) (length (unMaxLen b)) of
```

Comment:
If `MaxLen` will store `ByteString` instead of `String`, the `length` function will run in `O(1)` time instead of `O(n)`. Also, you can use `>=` operator here to reduce pattern matching to a single primitive operation 🙂